unlock-js/src/__tests__/deploy.test.js

```diff
@@ -114,29 +40,46 @@ describe('contract deployer', () => {
     status: '0x1',
   }
   beforeEach(() => {
-    accountsAndYield(unlockAccountsOnNode)
-    ethGasPriceAndYield()
-    sendTransactionAndYield(unlockAccountsOnNode[0], transaction.hash)
-    ethGetTransactionReceiptAndYield(transaction.hash, transactionReceipt)
-    ethGetCodeAndYield('0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', 'latest')
-    ethGasPriceAndYield()
-    sendTransactionAndYield(
-      unlockAccountsOnNode[0],
-      transaction.hash,
-      '0xc4d66de8000000000000000000000000aaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // data (no idea what this is, but there it is)
-      '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // contract address
-      '0xf4240' // gas
+    nock.accountsAndYield(unlockAccountsOnNode)
+    nock.ethGasPriceAndYield(gasPrice)
+    // contract deploy call
+    nock.ethSendTransactionAndYield(
+      {
+        from: unlockAccountsOnNode[0],
+        data: Unlock.bytecode,
+        gas: '0x' + GAS_AMOUNTS.deployContract.toString(16),
+      },
+      gasPrice,
+      transaction.hash
+    )
+    nock.ethGetTransactionReceipt(transaction.hash, transactionReceipt)
+    // get the contract bytecode
+    nock.ethGetCodeAndYield(
+      '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2',
+      'latest',
+      Unlock.bytecode
+    )
+    nock.ethGasPriceAndYield(gasPrice)
+    // initialize call
+    nock.ethSendTransactionAndYield(
+      {
+        to: '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // contract address
+        from: unlockAccountsOnNode[0],
+        data:
+          '0xc4d66de8000000000000000000000000aaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // data (no idea what this is, but there it is)
+        gas: '0xf4240',
+      },
+      gasPrice,
+      transaction.hash
     )
-    ethGetTransactionReceiptAndYield(transaction.hash, transactionReceipt)
+    nock.ethGetTransactionReceipt(transaction.hash, transactionReceipt)
   })
-  it('retrieves accounts', async () => {
-    expect.assertions(1)
-    const web3 = new Web3(`http://${host}:${port}`)
-    const getAccounts = web3.eth.getAccounts
-    web3.eth.getAccounts = jest.fn(() => getAccounts())
 
-    await deploy(host, port, Unlock, () => {}, web3)
-    expect(web3.eth.getAccounts).toHaveBeenCalled()
+  it('does not throw exception (json-rpc calls are accurate)', async () => {
```

Comment:
I am not sure it is great that we only have a single test left which is not really a test because everything happens in the `before` block.

We need to decompose what is actually happening...